contract/src/account.rs

```diff
@@ -172,11 +249,18 @@ impl Contract {
 
     pub fn internal_storage_balance_of(&self, account_id: &AccountId) -> Option<StorageBalance> {
         self.internal_get_account(account_id.as_str())
-            .map(|storage| StorageBalance {
-                total: storage.storage_balance.into(),
+            .map(|account| StorageBalance {
+                total: account.storage_balance.into(),
                 available: U128(
-                    storage.storage_balance
-                        - Balance::from(storage.used_bytes) * env::storage_byte_cost(),
+                    account.storage_balance
+                        - Balance::from(
+                            account.used_bytes
+                                - account
+                                    .shared_storage
+                                    .as_ref()
+                                    .map(|s| s.used_bytes)
+                                    .unwrap_or(0),
+                        ) * env::storage_byte_cost(),
```

Comment:
@evgenykuzyakov Is this math correct? I'm not able to save a widget after sharing a balance from a shared pool. After looking into VM and seeing that it uses `storage_balance_of => storage.available`, I think the math here looks odd.

Shouldn't it be something more like this:
```
available = (account.storage_balance - account.used_bytes * storage_byte_cost)
          + (shared.max_bytes - shared.used_bytes) * storage_byte_cost
```

Reply:
The reason why `storage_balance_of` implements it this way, because the standard requires available_storage to be withdrawable. There is a separate implementation with a different name `get_account_storage` that accounts for shared storage